recipes/yasm/all/patches/0001-platformtoolset.patch

```diff
@@ -0,0 +1,252 @@
+diff --git a/Mkfiles/vc10/genmacro/genmacro.vcxproj b/Mkfiles/vc10/genmacro/genmacro.vcxproj
+index 15cf1693..6c971f21 100644
+--- a/Mkfiles/vc10/genmacro/genmacro.vcxproj
++++ b/Mkfiles/vc10/genmacro/genmacro.vcxproj
+@@ -19,10 +19,12 @@
+   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|Win32'" Label="Configuration">
+     <ConfigurationType>Application</ConfigurationType>
+     <CharacterSet>MultiByte</CharacterSet>
++    <PlatformToolset>v142</PlatformToolset>
```

Comment:
🤔 https://github.com/conan-io/conan/blob/0ae90e8e63fef83486ef741a91108731450e028a/conan/tools/microsoft/toolchain.py#L38

The should be set by the new toolchain + build helper... maybe it's only the old compiler thats failing?

Reply:
> A conanvcvars.bat file with the necessary vcvars invocation to define the build environment if necessary to build from the command line or from automated tools (might not be necessary if opening the IDE). This file will be automatically called by the tools.microsoft.MSBuild helper build() method.

https://docs.conan.io/en/latest/reference/conanfile/tools/microsoft.html#msbuildtoolchain

Reply:
I don't know if this is the right way to do. I opened the `Mkfiles/vc10/yasm.sln` with Visual Studio 2019 and it asked me to upgrade the project which lead to this patch.

Maybe @madebr can help ?

I'm not familiar with MSBuild.

Reply:
https://github.com/conan-io/tribe/blob/main/design/032-msvc_support.md#motivation

Looking at the support is seems barely down to vc11 so I think the build helper just does not support this and slash or was never tested. In this release theres vc12 folder maybe we can try that?

If not in master they have newer vs build files we can use?

there's something missing with v2 if this needs to be added?